services/QuillLMS/app/workers/snapshots/cache_snapshot_count_worker.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+module Snapshots
+  class CacheSnapshotCountWorker
+    include Sidekiq::Worker
+
+    DEFAULT_CACHE_EXPIRATION = 24.hours
+    PUSHER_EVENT = 'admin-snapshot-count-cached'
+
+    QUERIES = {
+      'active-classrooms' => Snapshots::ActiveClassroomsQuery,
+      'active-students' => Snapshots::ActiveStudentsQuery,
+      'activities-assigned' => Snapshots::ActivitiesAssignedQuery,
+      'activities-completed' => Snapshots::ActivitiesCompletedQuery,
+      'average-activities-completed-per-student' => Snapshots::AverageActivitiesCompletedPerStudentQuery,
+      'sentences-written' => Snapshots::SentencesWrittenQuery,
+      'student-learning-hours' => Snapshots::StudentLearningHoursQuery
+    }
+
+    def perform(query, user_id, timeframe, school_ids, grades)
+      timeframe_name = timeframe[:name]
+      previous_timeframe_start = timeframe[:previous_start]
+      current_timeframe_start = timeframe[:current_start]
+      timeframe_end = timeframe[:current_end]
+
+      current_snapshot = QUERIES[query].run(user_id,
+        current_timeframe_start,
+        timeframe_end,
+        school_ids,
+        grades)
+
+      previous_snapshot = QUERIES[query].run(user_id,
+        previous_timeframe_start,
+        current_timeframe_start,
+        school_ids,
+        grades)
+
+      payload = { current: current_snapshot, previous: previous_snapshot }
+
+      cache_key = Snapshots::CacheKeys.generate_key(query,
+        user_id,
+        {
+          name: timeframe_name,
+          custom_start: timeframe_name == Snapshots::CacheKeys::CUSTOM_TIMEFRAME_NAME ? current_timeframe_start : nil,
+          custom_end: timeframe_name == Snapshots::CacheKeys::CUSTOM_TIMEFRAME_NAME ? current_timeframe_end : nil
+        },
+        school_ids,
+        grades)
+
+      Rails.cache.write(cache_key, payload, expires_in: DEFAULT_CACHE_EXPIRATION)
+
+      PusherTrigger.run(user_id, PUSHER_EVENT,
```

Comment:
After touching base with Emilia she raised a valid point: Pusher has a payload limit of 10KB.  While the values we're expecting in these snapshot returns, so it shouldn't be an issue, a number of our other reports easily exceed this limit.  This suggests that maintaining the existing pattern is worth doing since there are some cases where the second alternative you mention isn't an option, but the former alternative should always work.